### PR TITLE
Add GEOMETRY type e2e test suite

### PR DESCRIPTION
**Pull request by sfc-gh-pczajka**
## Summary
- Add comprehensive e2e tests for GEOMETRY type: type casting, literal selects, NULL handling, table operations, multi-chunk downloads, parameter binding, and JSON result format
- Add `parse_geojson()` helper to `utils.py` for GeoJSON string parsing
- Add `geometry.feature` Gherkin definitions (11 scenarios)

## Test plan
- [ ] All 11 GEOMETRY tests pass on universal driver (currently blocked by pre-existing protobuf bug)
- [ ] All 11 GEOMETRY tests pass on reference driver
- [ ] Test format validator passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/definitions/shared/types/geometry.feature [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    When Query "SELECT * FROM <table> ORDER BY id" is executed
+    Then Result should contain [GeoJSON Point, NULL]
+
+  # =========================================================================== #
+  #                       Multiple chunks downloading                           #
+  # =========================================================================== #
+
+  @python_e2e
+  Scenario: should download geometry data in multiple chunks
+    # skip_for_json_result_set
```
Comment (sfc-gh-asolarski) [inline tests/definitions/shared/types/geometry.feature:70]: we don't add it in scenarios in all other cases

File: python/tests/e2e/types/test_geometry.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+# Simple rectangle polygon (ring must be closed)
+POLYGON_WKT = "POLYGON((0 0, 4 0, 4 3, 0 3, 0 0))"
+POLYGON_GEOJSON_COORDS = [[[0.0, 0.0], [4.0, 0.0], [4.0, 3.0], [0.0, 3.0], [0.0, 0.0]]]
+
+# =============================================================================
+# LARGE RESULT SET SIZE
+# =============================================================================
+LARGE_RESULT_SET_SIZE = 20_000
+
+SKIP_JSON = pytest.mark.skip_for_json_result_set(reason="GEOMETRY type is not supported in JSON result format")
```
Comment (sfc-gh-asolarski) [inline python/tests/e2e/types/test_geometry.py]: if we skip all in the file, then we don't really need this decorator

Comment (sfc-gh-asolarski) [reply python/tests/e2e/types/test_geometry.py]: I mean - we don't need to apply it to all classes

**Review by sfc-gh-asolarski — APPROVED** [review]
